docs: update k8s-microk8s.md (#2431)
splunk/splunk-connect-for-syslog: docs/gettingstarted/k8s-microk8s.md

# Install and configure SC4S with Kubernetes
Splunk provides an implementation for SC4S deployment with MicroK8s using a single-server MicroK8s as the deployment model. Clustering has some tradeoffs and should be only considered on a deployment-specific basis.

> You can independently replicate the model deployment on different distributions of Kubernetes. To do this you must have Administrator privleges, and the process will require advanced understanding and responsibility on your part.

Reviewer: ```suggestion
You can independently replicate the model deployment on different distributions of Kubernetes. To do this you must have Administrator privileges, and the process will require advanced understanding and responsibility on your part.
```

I think the original piece mean that the administrator is expected to have advanced awareness and responsibility, not that Administrator privileges are required

Author: ah! How about something like this? "Do not attempt this unless you have advanced understanding of Kubernetes and are willing and able to maintain this configuration regularly."

Reviewer: yess, that's perfect

> You can independently replicate the model deployment on different distributions of Kubernetes. To do this you must have Administrator privleges, and the process will require advanced understanding and responsibility on your part.

Reviewer: I think the original piece mean that the administrator is expected to have advanced awareness and responsibility, not that Administrator privileges are required

microk8s enable openebs 
microk8s enable helm3
microk8s status --wait-ready
```

## Step 4: Add an SC4S Helm repository
To add an SC4S Helm repository:

```bash
microk8s helm3 repo add splunk-connect-for-syslog https://splunk.github.io/splunk-connect-for-syslog
microk8s helm3 repo update
```

## Step 5: Create a `values.yaml` file
Create the configuration file `values.yaml`. You can provide HEC token as a Kubernetes secret or in plain text. 

### Provide the HEC token as plain text
1. Create `values.yaml` file:

```yaml
--8<---- "docs/resources/k8s/values_basic.yaml"
```

2. Install SC4S:
```bash
microk8s helm3 install sc4s splunk-connect-for-syslog/splunk-connect-for-syslog -f values.yaml
```

### Provide the HEC token as secret
1. Create `values.yaml` file:

```yaml
--8<---- "docs/resources/k8s/values_basic_no_token.yaml"
```

2. Install SC4S:
```bash
export HEC_TOKEN="00000000-0000-0000-0000-000000000000"
microk8s helm3 install sc4s --set splunk.hec_token=$HEC_TOKEN splunk-connect-for-syslog/splunk-connect-for-syslog -f values.yaml
```

# Update or upgrade SC4S 

> Whenever the image is upgraded or when changes are made to the `values.yaml` file should be applied, run the command:

Reviewer: ```suggestion
Whenever the image is upgraded or when changes are made to the `values.yaml` file and should be applied, run the command:
```

1. Update the configuration file:
```yaml
--8<---- "docs/resources/k8s/values_ha.yaml"
```

2. Upgrade SC4S to apply the new configuration:
```bash
microk8s helm3 upgrade sc4s splunk-connect-for-syslog/splunk-connect-for-syslog -f values.yaml
```

# Configure your SC4S instances through `values.yaml`

With helm-based deployment you cannot configure environment variables and 
context files directly. Instead, use the `values.yaml` file to update your configuration, for example:

```yaml
--8<---- "docs/resources/k8s/values_adv.yaml"

```

Use the `config_files` and `context_files` variables to specify configuration and context files that are passed to SC4S.

- `config_files`: This variable contains a dictionary that maps the name of the configuration file to its content in the form of a YAML block scalar.
- `context_file`: This variable contains a dictionary that maps the name of the context files to its content in the form of a YAML block scalar. The context files `splunk_metadata.csv` and `host.csv` are passed with `values.yaml`:
```yaml
--8<---- "docs/resources/k8s/values_adv_config_file.yaml"
```

# Manage resources

> Provision two instances per node. Adjust requests and limits to allow each instance to use about 40% of each node, presuming no other workload is present.

Reviewer: What we mean here is that by default the user should expect two instances per node so they should assign about 40% of resource usage to each instance on every node, unless there is other workload. So maybe sth like:

```suggestion
You should expect two instances per node by default, so adjust requests and limits to allow each instance to use about 40% of each node, presuming no other workload is present.
```

Author: How about this: "You should expect your system to require two instances per node by default. Adjust requests and limits to allow each instance to use about 40% of each node, presuming no other workload is present. "

Reviewer: great
